jsdoc/services/transforms/extract-module-name.spec.js

```diff
@@ -0,0 +1,58 @@
+var Dgeni = require('dgeni');
+var mockPackage = require('../../mocks/mockPackage');
+
+describe("extract-module-name transform", function() {
+  var doc, transform;
+
+  beforeEach(function() {
+    var dgeni = new Dgeni([mockPackage()]);
+    var injector = dgeni.configureInjector();
+
+    transform = injector.get('extractModuleNameTransform');
+    doc = {
+      fileInfo: { projectRelativePath: 'project/relative/path.js' },
+      codeNode: {},
+      codeAncestors: {}
+    };
+  });
+
+  it("should extract module name without prefix", function() {
+    var tag = { tagName: 'module' }, value = 'customModuleName';
+    var newValue = transform(doc, tag, value);
+
+    expect(newValue).toEqual('');
+    expect(tag).toEqual({ tagName: 'module', name: 'customModuleName' });
+    expect(doc.codeNode).toBeNull();
+    expect(doc.codeAncestors).toBeNull();
+  });
+
+  it("should extract module name with prefix", function() {
+    var tag = { tagName: 'module' }, value = 'module:customName';
+    var newValue = transform(doc, tag, value);
+
+    expect(newValue).toEqual('');
+    expect(tag).toEqual({ tagName: 'module', name: 'module:customName' });
```

Comment:
From my reading of the [JSDOC docs](https://jsdoc.app/tags-module.html) the name in this case should just be `customName`. I.e. the `module:` prefix is not part of the name.